flask_test_app/app.py

```diff
@@ -0,0 +1,116 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Flask application made for integration tests."""
+
+import time
+from urllib.parse import urlparse
+
+import psycopg
+import pymysql
+import pymysql.cursors
+from flask import Flask, g, jsonify, request
+
+app = Flask(__name__)
+app.config.from_prefixed_env()
+
+
+def get_mysql_database():
+    """Get the mysql db connection."""
+    if "mysql_db" not in g:
+        if app.config.get("MYSQL_DB_CONNECT_STRING"):
+            uri_parts = urlparse(app.config["MYSQL_DB_CONNECT_STRING"])
+            g.mysql_db = pymysql.connect(
+                host=uri_parts.hostname,
+                user=uri_parts.username,
+                password=uri_parts.password,
+                database=uri_parts.path[1:],
+                port=uri_parts.port,
+            )
+        else:
+            return None
+    return g.mysql_db
+
+
+def get_postgresql_database():
```

Comment:
Should this code comply with the contributing guide (e.g., type hints, docstrings etc)?

Reply:
It hadn't before so I wasn't too focused on that. I can take some time for that

Reply:
It is starting to be a bit more complex which is when type hints can start to become helpful, though that can be mitigated using a library like flask-mysql to reduce the complexity

Reply:
I think this is a discussion point that deserves some attention. Do we want to have type hints to help us maintain that test application or do we want to have type hints to say that we feel that with a certain degree of complexity an app needs to have type hints? With our code (like src/charm.py) the answer is clear, we type hint everything and that's good. For the test app, it might serve it's purpose without type hints as initially created. I'm not sure whether it will get to a really complex state, being a test app that is. Even `charmcraft init` will create you a (test) charm that doesn't have any type hints and leaves it to you, we might want to do the same (or not). Complexity is also a bit subjective, so again - let's have the discussion for now and then see how to tackle this (with an additional PR).